### ISSUE TITLE
Duplicate dots generation

### ISSUE THREAD

**Issue by shubanms** [shubanms/dot-knot#1]
# Issue

Below function generates duplicate dot locations in array and places them on the grid

```python
 def generate_dots_positions(self):
```

**Comment by justforfunweb** [comment]
You could add a `if` check statement for an array of dots generated and take the ones which don't contain duplicates

```python
def generate_dots_positions(self):
```

In this loop 
```python
for _ in range(self.dots):
    dot_placements = []
```

under the function 
```python
def generate_dots_positions(self):
```

add a if check after you call the function `def generate_dots_positions(self):`

**Comment by shubanms** [comment]
> You could add a `if` check statement for an array of dots generated and take the ones which don't contain duplicates
> 
> ```python
> def generate_dots_positions(self):
> ```
> 
> In this loop
> 
> ```python
> for _ in range(self.dots):
>     dot_placements = []
> ```
> 
> under the function
> 
> ```python
> def generate_dots_positions(self):
> ```
> 
> add a if check after you call the function `def generate_dots_positions(self):`

Could you please create a pull request with said changes @justforfunweb

**Comment by shubanms** [comment]
Fixed by @justforfunweb in PR #4
